src/Val.h

```diff
@@ -1095,6 +1092,9 @@ class TableVal final : public Val, public notifier::detail::Modifiable {
 	// Sends data on to backing Broker Store
 	void SendToStore(const Val* index, const TableEntryVal* new_entry_val, OnChangeType tpe);
 
+	// Low-level insertion, not publicly available.
+	void Insert(detail::HashKey* k, TableEntryVal* tev);
```

Comment:
Let's aim to switch to smart ptrs for new methods passing dynamic instances around.

Reply:
What would be appropriate here?  Neither `HashKey` nor `TableEntryVal` are derived from `Obj`, so they don't fit with the `IntrusivePtr` framework.  Note that all that method does is `table_val->Insert(k, tev);`.  I could just change its callers to do that directly, now that it's an internal method.

Reply:
As it's passing ownership of `k` and `tev`, `unique_ptr<>` would express that and ensure it's done correctly. But yeah, if we don't really need it anyways, skipping this sounds better.

Generally, the standard `shared_ptr`/`unique_ptr` are usually good choices for anything dynamically allocated. It's `IntrusivePtr` that's the exception to that rule: We need that only for `Obj`-derived classes because they already do their memory management internally, so `shared_ptr`/`unique_ptr` wouldn't work.